packages/synapse-sdk/src/storage/context.ts

```diff
@@ -1260,6 +1260,28 @@ export class StorageContext {
     return dataSetData.pieces.map((piece) => piece.pieceCid)
   }
 
+  /**
+   * Delete a piece with given CID from this data set
+   * @param pieceCid - The PieceCID identifier
+   * @returns Transaction hash of the delete operation
+   */
+  async deletePiece(pieceCid: string | PieceCID): Promise<string> {
```

Comment:
hey, this could be:


```suggestion
  async deletePiece(piece: string | PieceCID | number): Promise<string> {
```

and if it's a `number` then we skip the pieceId look-up and just use that.

Then, if we do https://github.com/FilOzone/synapse-sdk/pull/249#issuecomment-3379605263 and expose a `getPieces(): AsyncGenerator<{ cid: PieceCID, id: number }>` on the context, the user can be in charge of choosing which piece to delete and if there's duplicates they can pinpoint exactly which one.

Reply:
Yup, I've implemented accepting the pieceID